### ISSUE TITLE
I can't access hidden .onion site on my tor browser.

### ISSUE THREAD

**Issue by EnterUsername** [xkollar/tor-openshift#2]
Am I missing something?
My step:

```
rhc app create "home" --no-git diy 'https://raw.githubusercontent.com/xkollar/tor-openshift/master/metadata/manifest.yml' --from-code 'https://github.com/xkollar/openshiftor-example.git'
```

My app:
https://home-shadowcrew.rhcloud.com/

**Comment by xkollar** [comment]
I will try to answer your question in at most finite time.

EDIT: made comment more accurate (sorry for the delay).

**Comment by EnterUsername** [comment]
Thank you for your support.

**Comment by EnterUsername** [comment]
Tor fails to start.
**tor.log**   does not exist in   **${OPENSHIFT_LOG_DIR}**
My **tor.install.log** file says that tor fails with the following problem:

```
  ...      ... ...
  ...      ... ...
  ...      ... ...
  GEN      doc/tor.1
  GEN      doc/tor-gencert.1
  GEN      doc/tor-resolve.1
  GEN      doc/torify.1
  GEN      doc/tor.html
  GEN      doc/tor-gencert.html
  GEN      doc/tor-resolve.html
  GEN      doc/torify.html
make[1]: Leaving directory `/tmp/tmp.GXlIoZF5fx/tor-0.2.6.10'
make[1]: Entering directory `/tmp/tmp.GXlIoZF5fx/tor-0.2.6.10'
 /bin/mkdir -p '/var/lib/openshift/560f07a67628e17cee000098/tor/tor/bin'
  /usr/bin/install -c src/or/tor src/tools/tor-resolve src/tools/tor-gencert '/var/lib/openshift/560f07a67628e17cee000098/tor/tor/bin'
 /bin/mkdir -p '/var/lib/openshift/560f07a67628e17cee000098/tor/tor/bin'
 /usr/bin/install -c contrib/client-tools/torify '/var/lib/openshift/560f07a67628e17cee000098/tor/tor/bin'
 /bin/mkdir -p '/var/lib/openshift/560f07a67628e17cee000098/tor/tor/etc/tor'
 /usr/bin/install -c -m 644 src/config/torrc.sample '/var/lib/openshift/560f07a67628e17cee000098/tor/tor/etc/tor'
 /bin/mkdir -p '/var/lib/openshift/560f07a67628e17cee000098/tor/tor/share/doc/tor'
 /usr/bin/install -c -m 644 doc/tor.html doc/tor-gencert.html doc/tor-resolve.html doc/torify.html '/var/lib/openshift/560f07a67628e17cee000098/tor/tor/share/doc/tor'
 /bin/mkdir -p '/var/lib/openshift/560f07a67628e17cee000098/tor/tor/share/man/man1'
 /usr/bin/install -c -m 644 doc/tor.1 doc/tor-gencert.1 doc/tor-resolve.1 doc/torify.1 '/var/lib/openshift/560f07a67628e17cee000098/tor/tor/share/man/man1'
 /bin/mkdir -p '/var/lib/openshift/560f07a67628e17cee000098/tor/tor/share/tor'
 /usr/bin/install -c -m 644 src/config/geoip src/config/geoip6 '/var/lib/openshift/560f07a67628e17cee000098/tor/tor/share/tor'
make[1]: Leaving directory `/tmp/tmp.GXlIoZF5fx/tor-0.2.6.10'
/tmp/tmp.GXlIoZF5fx
Oct 02 18:48:53.460 [notice] Tor v0.2.6.10 (git-58c51dc6087b0936) running on Linux with Libevent 1.4.13-stable, OpenSSL 1.0.1e-fips and Zlib 1.2.3.
Oct 02 18:48:53.460 [notice] Tor can't help you if you use it wrong! Learn how to be safe at https://www.torproject.org/download/download#warning
Oct 02 18:48:53.460 [notice] Read configuration file "/var/lib/openshift/560f07a67628e17cee000098/tor/tor/etc/tor/torrc".
Oct 02 18:48:53.468 [warn] Permissions on directory /var/lib/openshift/560f07a67628e17cee000098/tor/tor/hidden_service/ are too permissive.
Oct 02 18:48:53.468 [warn] Failed to parse/validate config: Failed to configure rendezvous options. See logs for details.
Oct 02 18:48:53.468 [err] Reading config failed--see warnings above.
"tor.install.log" 1229L, 54620C                               1222,1        Bot
```

**Comment by xkollar** [comment]
Ok, some compulsory comments. Do not (try to) run tor service on any free hosting service without familiarizing yourself with their terms&conditions. They face many attempts for misuse/abuse and it is easy for them to check for Tor. Ask for permission, explain what you are trying to achieve. You might get permission if you explain your case well. Do not betray gained trust.

Also: It take some time for Tor to setup circuits and it also takes some time for rendezvous points to get propagated properly, so be patient.

Also: Thanks for the report :+1:.

**Comment by EnterUsername** [comment]
Thanks.
Worked with Tor-0.2.5.12

or

Tor-0.2.6.10  with  ' chmod 700 ${OPENSHIFT_TOR_DIR}tor/hidden_service/ '
http://stackoverflow.com/questions/31620783/what-is-wrong-with-my-tor

**Comment by xkollar** [comment]
Mentioned patch (d80aca30d17bff4dd1d917e24c31f036d1dc430e) fixed it. And yes, it was just adding `chmod`. (To be honest, I have tested it only in mocked environment, not on the OpenShift.)

**Comment by EnterUsername** [comment]
Yes. Problem already be fixed in [d80aca3](https://github.com/xkollar/tor-openshift/commit/d80aca30d17bff4dd1d917e24c31f036d1dc430e)
I have tested it on the OpenShift.
I just want to host my contact info (static html) to OpenShift with an onion address.
This project would be perfect for me :hatched_chick:
Thank you :+1: .
